R/check_soft_duplicates.R

```diff
@@ -0,0 +1,100 @@
+#' Checks for survey similarities - Soft Duplicates
+#'
+#' @param dataset RawData to be checked as a dataframe or a list with dataframe
+#'  stored as "checked_dataset"
+#' @param kobo_survey Kobo Survey Sheet
+#' @param uuid String character for the uuid column. By default "uuid"
+#' @param idnk_value String character for the value of the "I don't know" value
+#' @param sm_seperator Separator for choice multiple questions. The default is "."
+#' @param log_name Name of the log dataframe flagged in the end of the function
+#'
+#' @return return a list with the dataset checked stored as checked_dataset and
+#' a dataframe with the soft duplicate log
+#' @export
+#'
+#' @examples
+#' soft_duplicates <- check_soft_duplicates(dataset = cleaningtools_raw_data,
+#'                                          kobo_survey = cleaningtools_survey,
+#'                                          uuid = "X_uuid",
+#'                                          idnk_value = "dont_know",
+#'                                          sm_seperator = ".",
+#'                                          log_name = "soft_duplicate_log")
+
+check_soft_duplicates <- function(dataset,
+                                  kobo_survey,
+                                  uuid="uuid",
+                                  idnk_value="idnk",
+                                  sm_seperator = ".",
+                                  log_name = "soft_duplicate_log"){
+
+  if(is.data.frame(dataset)){
+    dataset <- list(checked_dataset = dataset)
+  }
+
+  if(!"checked_dataset" %in% names(dataset)) {
+    stop("Cannot identify the dataset in the list.")
+  }
+
+  if(!uuid %in% names(dataset[["checked_dataset"]])){
+    msg <- glue::glue("Cannot find ", uuid, " in the names of the dataset")
+    stop(msg)
+  }
+
+  data <- dataset$checked_dataset
+  # 1) store UUIDs
+  uuids <- data[[uuid]]
+
+  # 2) convert all columns to character and tolower
+  data <- dplyr::mutate_all(data, as.character)
+  data <- dplyr::mutate_all(data, tolower)
+
+  # 3) remove columns that are naturally different in each survey:
```

Comment:
![image](https://github.com/impact-initiatives/cleaningtools/assets/79831444/a97b0eb7-703b-4ba4-a293-ab3b89d334be)
maybe having a filter!is.na(type) after the left_join would remove a few of the missing, like starts with X_, punct